### PR TITLE
Added benchmark module

### PR DESCRIPTION
**Pull request by segabriel**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -87,6 +87,7 @@
     <module>services-codec</module>
     <module>services-codec-jackson</module>
     <module>services</module>
+    <module>benchmark</module>
```
Comment (artem-v) [inline pom.xml]: `benchmarks`, plural

**Review by artem-v — CHANGES_REQUESTED** [review]

File: benchmark/pom.xml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>scalecube-services-parent</artifactId>
+        <groupId>io.scalecube</groupId>
+        <version>2.0.16-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>scalecube-services-benchmark</artifactId>
+    <name>ScaleCube/scalecube-services-benchmark</name>
```
Comment (artem-v) [inline benchmark/pom.xml]: `benchmarks` , plural

File: benchmark/pom.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>scalecube-services-parent</artifactId>
+        <groupId>io.scalecube</groupId>
+        <version>2.0.16-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>scalecube-services-benchmark</artifactId>
```
Comment (artem-v) [inline benchmark/pom.xml]: `benchmarks` , plural

File: benchmark/src/main/java/io/scalecube/services/benchmark/transport/FireAndForgetBenchmark.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+@Measurement(iterations = 2, time = 5)
+public class FireAndForgetBenchmark {
+
+  private static final SimpleBenchmarkRequest REQUEST = new SimpleBenchmarkRequest("monoVoidServiceBenchmark");
+
+  private Microservices seed;
+  private Microservices node;
+  private BenchmarkService benchmarkService;
+
+  @Setup(Level.Trial)
```
Comment (artem-v) [inline benchmark/src/main/java/io/scalecube/services/benchmark/transport/FireAndForgetBenchmark.java]: Setup comes with default level trial , no need to set it explicitly

File: benchmark/src/main/java/io/scalecube/services/benchmark/transport/FireAndForgetBenchmark.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.util.concurrent.TimeUnit;
+
+@Fork(2)
+@State(Scope.Benchmark)
+@Threads(4)
+@Warmup(iterations = 2)
+@Measurement(iterations = 2, time = 5)
```
Comment (artem-v) [inline benchmark/src/main/java/io/scalecube/services/benchmark/transport/FireAndForgetBenchmark.java]: `, time = 5`  what's `time` ?

**Review by artem-v — CHANGES_REQUESTED** [review]

File: benchmark/src/main/java/io/scalecube/services/benchmark/transport/FireAndForgetBenchmark.java (hunk 5 above)
Comment (segabriel) [reply benchmark/src/main/java/io/scalecube/services/benchmark/transport/FireAndForgetBenchmark.java]: time is a value of seconds (default unit time), and it's 10 by default.

File: pom.xml [hunk 6]
```diff
@@ -60,10 +60,10 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <jackson.version>2.9.0</jackson.version>
     <scalecube.version>1.0.9</scalecube.version>
-    <reactor.version>3.1.6.RELEASE</reactor.version>
+    <reactor.version>3.1.7.RELEASE</reactor.version>
     <rxjava.version>1.3.8</rxjava.version>
     <metrics.version>3.1.2</metrics.version>
-    <rsocket.version>0.11.1</rsocket.version>
+    <rsocket.version>0.11.3</rsocket.version>
```
Comment (artem-v) [inline pom.xml:66]: **NOTE**: `io.scalecube.services.benchmarks.LeakDetectorRunner` complained about buffer leaks in rSocket 0.11.1. In .3 version they fixed this buffer leak (look at maven central how many releases rSocket team made on May 2018), hence here's update.

File: pom.xml [hunk 7]
```diff
@@ -60,10 +60,10 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <jackson.version>2.9.0</jackson.version>
     <scalecube.version>1.0.9</scalecube.version>
-    <reactor.version>3.1.6.RELEASE</reactor.version>
+    <reactor.version>3.1.7.RELEASE</reactor.version>
```
Comment (artem-v) [inline pom.xml:63]: This update just came along naturally.

**Review by artem-v — CHANGES_REQUESTED** [review]

**Pull request merged**